processor/src/chiplets/hasher/mod.rs

```diff
@@ -500,6 +563,31 @@ impl Hasher {
             _ => (),
         }
     }
+
+    /// Checks if a trace for a program block already exists and returns the start and end rows
+    /// of the memoized trace. Returns None otherwise.
+    fn get_memoized_trace(&self, hash: Digest) -> Option<&(usize, usize)> {
+        let hash_word: Word = hash
+            .as_elements()
+            .try_into()
+            .expect("Could not convert digest to word");
+        let key = hash_word.into_bytes();
+        self.memoized_trace_map.get(&key)
+    }
+
+    /// Inserts start and end rows of trace for a program block to the memoized_trace_map.
+    fn insert_to_memoized_trace_map(&mut self, addr: Felt, hash: Digest) {
+        let hash_word: Word = hash
+            .as_elements()
+            .try_into()
+            .expect("Could not convert digest to word");
+        let key = hash_word.into_bytes();
+        let start_row = usize::try_from(addr.as_int()).expect("Could not convert u64 to usize") - 1;
+        let end_row = usize::try_from(self.trace.next_row_addr().as_int())
+            .expect("Could not convert u64 to usize")
+            - 1;
```

Comment:
nit: you should be able to simplify this as:
```Rust
let key: [u8; 32] = hash.into();
let start_row = addr.as_int() as usize - 1;
let end_row = self.trace.next_row_addr().as_int() as usize - 1;
```

Reply:
Fixed, thanks.